modules/effects/src/main/scala/dev/profunktor/redis4cats/interpreter/Redis.scala

```diff
@@ -223,10 +223,23 @@ private[redis4cats] class BaseRedis[F[_]: ContextShift, K, V](
       async.flatMap(c => F.delay(c.set(key, value)))
     }.void
 
-  override def set(key: K, value: V, setArgs: SetArgs): F[Boolean] =
+  override def set(key: K, value: V, setArgs: SetArgs): F[Boolean] = {
+    val jSetArgs = new JSetArgs()
+
+    setArgs.existence.foreach {
+      case SetArg.Existence.Nx => jSetArgs.nx()
```

Comment:
Is `JSetArgs` mutable? :open_mouth: 

Judging by the working tests I guess it is but would like to double-check, otherwise we would be discarding values here.

Reply:
yes, mutable :upside_down_face:

Reply:
Java stuff hehe. No worries, at least we keep the mutation local so the function remains referentially transparent.